lib/rspec/support/object_formatter.rb

```diff
@@ -88,6 +88,21 @@ def pretty_print(pp)
           pp.text inspection
         end
       end
+
+      # @private
+      DelegatingInspector = Struct.new(:object) do
+        def inspect
+          if defined?(::Delegator) && ::Delegator === object
+            "#<#{object.class}(#{DelegatingInspector.new(object.__getobj__).inspect})>"
+          else
+            object.inspect
+          end
+        end
+
+        def pretty_print(pp)
+          pp.text inspect
+        end
```

Comment:
Is there a spec that requires this method to pass?  I think you'd have to add a spec to the differ for it.  See here for an example:

https://github.com/rspec/rspec-support/blob/400b69dd70f1c1321f82ccabd882f144a610c1eb/spec/rspec/support/differ_spec.rb#L264-L320